src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -35,40 +41,55 @@ public ActiveSolutionBoundTracker(IHost host, IActiveSolutionTracker activeSolut
             this.extensionHost = host;
             this.solutionTracker = activeSolutionTracker;
 
-            this.extensionHost.VisualStateManager.BindingStateChanged += this.OnBindingStateChanged;
-            this.solutionTracker.ActiveSolutionChanged += this.OnActiveSolutionChanged;
+            this.solutionBindingInformationProvider = this.extensionHost.GetService<ISolutionBindingInformationProvider>();
+            this.solutionBindingInformationProvider.AssertLocalServiceIsNotNull();
 
             this.errorListInfoBarController = this.extensionHost.GetService<IErrorListInfoBarController>();
             this.errorListInfoBarController.AssertLocalServiceIsNotNull();
+            this.errorListInfoBarController.Refresh();
 
-            this.CalculateSolutionBinding();
-        }
+            // The user changed the binding through the Team Explorer
+            this.extensionHost.VisualStateManager.BindingStateChanged += this.OnBindingStateChanged;
+
+            // The solution changed inside the IDE
+            this.solutionTracker.ActiveSolutionChanged += this.OnActiveSolutionChanged;
 
-        public bool IsActiveSolutionBound { get; private set; } = false;
+            this.isSolutionPreviouslyBound = this.IsActiveSolutionBound;
+        }
 
         private void OnActiveSolutionChanged(object sender, EventArgs e)
         {
-            this.CalculateSolutionBinding();
+            this.RaiseAnalyzersChangedIfBindingChanged();
+            this.errorListInfoBarController.Refresh();
         }
 
         private void OnBindingStateChanged(object sender, EventArgs e)
         {
-            this.CalculateCurrentBinding();
+            this.RaiseAnalyzersChangedIfBindingChanged();
         }
 
-        private void CalculateSolutionBinding()
+        private void RaiseAnalyzersChangedIfBindingChanged()
         {
-            this.CalculateCurrentBinding();
-            this.errorListInfoBarController.Refresh();
+            if (this.isSolutionPreviouslyBound == this.IsActiveSolutionBound)
+            {
+                return;
+            }
+
+            this.isSolutionPreviouslyBound = this.IsActiveSolutionBound;
```

Comment:
Similarly to the constructor this should be the other way around - the property needs to be set based to bound/not bound.